### PR TITLE
Misc. bug fixes

### LINKED ISSUE #4578: Updating parameters without default values
Hi, 

As per my previous issue posted today, I want to start contributing to Pybamm, but I have never contributed to an open-source project before, so I am looking for an easy entry point to understand the process. 

A second point came up in some work I was doing today: when trying to update a parameter that doesn't have a default value, the following error is given:
` "Cannot update parameter '{name}' as it does not have a default value. ({err.args[0]}). If you are sure you want to update this parameter, use param.update({{name: value}}, check_already_exists=False)"
                  `

However, if I follow this advice and use param.update({{name: value}} where param is a ParameterValues object, I get the following error, 
TypeError: unhashable type: 'dict'

Perhaps this is just a typo? If I remove one set of curly brackets it works fine - param.update({name: value}, check_already_exists=False)

Again, just looking for a simple fix to make my first pull request. Please let me know what you think of this and if I have missed something.

### LINKED ISSUE #4580: [Bug]: Error thrown when not connected to the internet
### PyBaMM Version

develop

### Python Version

all

### Describe the bug

Due to the telemetry the following error is thrown when users are not connected to the internet:
```shell
error uploading: HTTPSConnectionPool(host='us.i.posthog.com', port=443): Max retries exceeded with url: /batch/ (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x15e8bfce0>: Failed to resolve 'us.i.posthog.com' ([Errno 8] nodename nor servname provided, or not known)"))
```

### Steps to Reproduce

This can be re-produced by just disabling your internet connection when first importing PyBaMM with telemetry enabled.

### PR DESCRIPTION
**Pull request by kratman**
# Description

A couple minor bug fixes before the release

Fixes #4580
Fixes #4578 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: src/pybamm/parameters/parameter_values.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ def update(self, values, check_conflict=False, check_already_exists=True, path="
                         f"Cannot update parameter '{name}' as it does not "
                         + f"have a default value. ({err.args[0]}). If you are "
                         + "sure you want to update this parameter, use "
-                        + "param.update({{name: value}}, check_already_exists=False)"
+                        + "param.update({name: value}, check_already_exists=False)"
```
Comment (kratman) [inline src/pybamm/parameters/parameter_values.py:246]: This was a mistake from when the f-strings were broken up

File: src/pybamm/__init__.py [hunk 2]
```diff
@@ -234,4 +235,4 @@
     "pybamm_data",
 ]
 
-pybamm.config.generate()
+config.generate()
```
Comment (kratman) [inline src/pybamm/__init__.py:238]: Doing it this way gets rid of an IDE warning

File: src/pybamm/telemetry.py [hunk 3]
```diff
@@ -3,12 +3,15 @@
...
 import sys
 
+
 _posthog = Posthog(
     # this is the public, write only API key, so it's ok to include it here
     project_api_key="phc_bLZKBW03XjgiRhbWnPsnKPr0iw0z03fA6ZZYjxgW7ej",
     host="https://us.i.posthog.com",
 )
 
+_posthog.log.setLevel("CRITICAL")
```
Comment (kratman) [inline src/pybamm/telemetry.py:12]: The error from not being connected to the internet comes out at the `ERROR` level. This stops the warning as far as I can tell

**Review by kratman — COMMENTED** [review]
Explanations

**Review by valentinsulzer — APPROVED** [review]

**Pull request merged**